rsocket-core/src/main/java/io/rsocket/RSocketServer.java

```diff
@@ -70,34 +69,19 @@
 
     connection
         .send(sendProcessor)
-        .doOnError(this::handleSendProcessorError)
         .doFinally(this::handleSendProcessorCancel)
-        .subscribe();
-
-    this.receiveDisposable =
-        connection
-            .receive()
-            .flatMapSequential(
-                frame ->
-                    handleFrame(frame)
-                        .onErrorResume(
-                            t -> {
-                              errorConsumer.accept(t);
-                              return Mono.empty();
-                            }))
-            .doOnError(errorConsumer)
-            .then()
-            .subscribe();
+        .subscribe(null, this::handleSendProcessorError);
+
+    this.receiveDisposable = connection.receive().subscribe(this::handleFrame, errorConsumer);
```

Comment:
This no long composes the backpressure from the client code and netty. That's why it's flap map - now it's disconnected because you have another subscription.